pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java

```diff
@@ -398,7 +398,7 @@ public Entry get() {
         public void readEntriesFailed(ManagedLedgerException exception, Object ctx) {
             if (managedLedger.getConfig().isAutoSkipNonRecoverableData()
                     && exception instanceof ManagedLedgerException.NonRecoverableLedgerException
-                    || exception instanceof ManagedLedgerException.ManagedLedgerFencedException) {
```

Comment:
Why remove the `ManagedLedgerFencedException` check?

Reply:
And from the current implementation, if return `isReadable` here which means the pending ack reply will be complete, is it expected behavior? https://github.com/apache/pulsar/pull/14781/files#diff-07a1d142ec4105cb65fac733494182b461214a68d735ae1c3f5104c3eb4f92dbL363

Reply:
> Why remove the `ManagedLedgerFencedException` check?

we can't delete ManagedLedgerFencedException, that's my mistake

Reply:
> And from the current implementation, if return `isReadable` here which means the pending ack reply will be complete, is it expected behavior? https://github.com/apache/pulsar/pull/14781/files#diff-07a1d142ec4105cb65fac733494182b461214a68d735ae1c3f5104c3eb4f92dbL363

if ManagedLedgerFencedException or CursorAlreadyClosedxception, the pending ack has been closed so we can stop the recover.